src/main/java/com/jcabi/ssh/SSH.java

```diff
@@ -215,7 +237,15 @@ protected Session session() throws IOException {
                 CharEncoding.UTF_8
             );
             jsch.setHostKeyRepository(new EasyRepo());
-            jsch.addIdentity(file.getAbsolutePath());
+            if (this.passphrase == null) {
```

Comment:
@LinkedList this null check said that something is definitely wrong with the design.
That's another story, it's jut FYI. If you have some ideas how to avoid these null checks - I'm all my ears